src/components/CreateManifestForm/CreateManifestForm.tsx

```diff
@@ -56,139 +50,138 @@ const CreateManifestForm: FC<CreateManifestFormProps> = (props) => {
     handleModalToggle();
   }
 
+  const satelliteTypeOptions = satelliteVersions?.map((satelliteVersion: SatelliteVersion) => {
+    return (
+      <FormSelectOption
+        isDisabled={false}
+        key={satelliteVersion.value}
+        value={satelliteVersion.value}
+        label={satelliteVersion.description}
+        validated={typeValidated}
+      />
+    );
+  });
+
+  const handleNameChange = (manifestName: string) => {
+    setNameFieldHelperText(
+      'Your manifest name must be unique and must contain only numbers, letters, underscores, and hyphens.'
+    );
+    setManifestName(manifestName);
+    setNameValidated('default');
+  };
+
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
+      if (
+        /^[0-9A-Za-z_.-]*$/.test(manifestName) &&
+        manifestName.length > 0 &&
+        manifestName.length < 99
+      ) {
+        setNameValidated('success');
+        setNameFieldHelperText(
+          'Your manifest name must be unique and must contain only numbers, letters, underscores, and hyphens.'
+        );
+      } else {
+        setNameValidated('error');
+        setInvalidNameFieldText(
+          'Name requirements have not been met. Your manifest name must be unique and must contain only numbers, letters, underscores, and hyphens.'
+        );
+      }
+    }, 500);
+
+    return () => clearTimeout(timer);
+  }, [manifestName]);
+
+  const typeSelectOnChange = (value: string, _event: React.FormEvent<HTMLSelectElement>) => {
+    setTypeValidated('success');
+    setManifestType(value);
+  };
+
+  React.useEffect(() => {
+    const timer = setTimeout(
+      (manifestType: string) => {
+        if (manifestType != '') {
+          setTypeValidated('success');
+        } else if (manifestType == '') {
+          setTypeValidated('error');
+          setInvalidTypeText('Selection Required');
+        }
+      },
+      500,
+      manifestType
+    );
```

Comment:
They both need timers, because according to the A/C and sketch design the form is supposed to render  in default state and after timeout illicit the invalid text indicators in each field. Without timeout it would immediately render those invalid text fields because of its default state.

Reply:
Hm - I'm not sure I follow. From the mockups I understood that the validation should only occur AFTER the respective field had been selected - this just delays it half a second from popping up but still does not wait for user input.

We should find some way to prevent this from running until that selection happens. Maybe wait for an `onClick` or something.

I just don't see the value in the half second delay - or the request for it in the AC or mockup

Reply:
I do see what your saying but my interpretation from the wireframe on sketch here " https://www.sketch.com/s/fce9da11-8245-4ff8-a4bc-2c63c1f04a75/a/Jn0waRz" it states to show errors when form loses focus with "no input" or when form is invalid. I'll add an amend to wait for action from user to make this more appropriate instead of timeout

Reply:
Ah I understand. We can confirm with UX but I think the intent of that sentence was actually "show errors with the input loses focus with 'no input' or when the input loses focus with invalid input" meaning no validation should run before anything loses focus

Reply:
It should be if you click into either input field, don't change anything, and then click out (lose focus) then the validation error should appear.

Also, if you click in the name field as soon as you input an invalid name (e.g. add $) then the validation error should appear.

Reply:
also, when i load the modal the validation errors are showing automatically..i think they should be hidden until there's an error state as described above.

![image](https://user-images.githubusercontent.com/11836261/204592943-1f987167-4f04-4351-84c4-285bd6a4b918.png)

Reply:
Ok I see now, I had a misinterpretation. It is clear now , I will make the changes.